mathesar/api/exceptions/mixins.py

```diff
@@ -0,0 +1,152 @@
+            if getattr(validator, 'requires_context', False):
+                args.append(field)
+            validator(self.initial_data[field.field_name], *args)
+        except (DjangoValidationError, RestValidationError) as err:
+            err_message = err.detail[0] \
+                if hasattr(err, 'detail') else err.message
+            return err_message == message
+
+    @property
+    def errors(self):
+        """
+        This method build on top of `errors` property of the superclass to return a list instead of a dictionary
+        """
+        ugly_errors = super(FriendlyErrorMessagesMixin, self).errors
+        pretty_errors = self.build_pretty_errors(ugly_errors)
+        return ReturnList(pretty_errors, serializer=self)
+
+    @property
+    def field_map(self):
+        """
+        This method build on top of `field_map` property of the superclass
+        It provides the following additional features
+        1. Adds `ListSerializer` to `relation` field list
+        """
+        return {
+            'boolean': ['BooleanField', 'NullBooleanField'],
+            'string': ['CharField', 'EmailField', 'RegexField', 'SlugField',
+                       'URLField', 'UUIDField', 'FilePathField',
+                       'IPAddressField'],
+            'numeric': ['IntegerField', 'FloatField', 'DecimalField'],
+            'date': {'DateTimeField': self.DATETIME_FORMAT,
+                     'DateField': self.DATE_FORMAT,
+                     'TimeField': self.TIME_FORMAT,
+                     'DurationField': self.DURATION_FORMAT},
+            'choice': ['ChoiceField', 'MultipleChoiceField'],
+            'file': ['FileField', 'ImageField'],
+            'composite': ['ListField', 'DictField', 'JSONField'],
+            'relation': ['StringRelatedField', 'PrimaryKeyRelatedField',
+                         'HyperlinkedRelatedField', 'SlugRelatedField',
+                         'HyperlinkedIdentityField', 'ManyRelatedField', 'ListSerializer'],
+            'miscellaneous': ['ReadOnlyField', 'HiddenField', 'ModelField',
+                              'SerializerMethodField']
+        }
+
+    def get_field_kwargs(self, field, field_data):
+        """
+        This method build on top of `get_field_kwargs` method of the superclass
+        It provides the following fixes
+        1. Fixes file type length value to use name of the file instead of the size of the file,
+         matching the default behaviour of drf
+        """
+        field_type = field.__class__.__name__
+        kwargs = {
+            'data_type': type(field_data).__name__
+        }
+        if field_type in self.field_map['boolean']:
+            kwargs.update({'input': field_data})
+        elif field_type in self.field_map['string']:
+            kwargs.update({'max_length': getattr(field, 'max_length', None),
+                           'min_length': getattr(field, 'min_length', None),
+                           'value': field_data})
```

Comment:
If you're splitting a dict into different lines, I prefer this:

```python
kwargs.update({
    'max_length': getattr(field, 'max_length', None),
    'min_length': getattr(field, 'min_length', None),
    'value': field_data
})
```

Also applies to other instances